src/optimizer/plan_generator.cpp

```diff
@@ -364,6 +375,7 @@ void PlanGenerator::Visit(const Limit *op) {
     }
 
     output_plan_ = order_build.Build();
+    plan_meta_data_->AddPlanNodeMetaData(output_plan_->GetPlanNodeId(), plan_node_meta_data_);
```

Comment:
Why we're explicitly adding the `PlanNodeMetaData` for `Limit` but not for others?

Reply:
Adding plan node meta data of other nodes is handled in `ConvertOpNode`. We need to do this for `Limit` in `Visit` because the limit node is not the `output_plan_` but an additional node generated in `Visit`.

Reply:
Got it! I think it's helpful to add comments to document this to avoid confusion.

Reply:
I still think it's good to add some comments in the code.

Reply:
I didn't see this comment. Yep, I'll add some.